### PR TITLE
Provide Time Slider to review events in previous turn

### LINKED ISSUE #2134: Provide Time Slider to review events in previous turn
Players will spend the morning planning what they want to happen.

The umpires will spend the afternoon deciding what actually happened.

The next morning, the players will see the new "state of the world", and want to review what happened.

All of that data is present in the "Planning" channel, but it's spread across different places.

I've just seen this demo:
https://skeate.dev/Leaflet.timeline/examples/earthquakes.html

It's a time-slider that moves through a series of timed GeoJSON features.  Each of our MessageInteraction messages contains a timed GeoSJON object.  We could just collate these into an array, and pass it to the `leaflet.timeline` control.

Ian can handle transforming the data, and getting it into the control, but he could do with support in integrating the control:
https://github.com/skeate/Leaflet.timeline

So, I guess the requirement is:
* [x] integrate control into Serge/components
* [x] introduce `showTimeSlider: boolean` and `timeSliderValues: GeoJSON.FeatureCollection` `useState` objects in `PlanningChannel`
* [x] add toggle button with `fa-history` icon to `PlanningChannel` to toggle the map in/out of `Replay` mode. Button toggles the value of `showTimeSlider`.
* [x] a new TimeSlider react component is produced, which takes a `events` prop as GeoJSON.FeatureCollection.
* [x] when `timeSlider` is true the timeline is shown at the bottom of the map
* [x] when `timeSlider` is true, a `useEffect` method will collate the GeoJSON.FeatureCollection data (I can do this :-) ), and store the array in `setTimeSliderValues`. The time-slider should now "wake-up", and we can move through the data.

Note: I was too excited, so I made a start here: https://github.com/serge-web/serge/pull/2135

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #2134 

https://user-images.githubusercontent.com/26705621/202908813-8e51ea5f-cfe4-41a6-a7c2-5859e823e385.mov

### CODE REVIEW

**Comment by IanMayo** [comment]
Handful of changes, please:
* [ ] Don't display blue-line to left of thumb-marker. There is guidance on how to do it [here](https://www.w3schools.com/howto/howto_js_rangeslider.asp)
* [x] I wish to color the features according to their Force. So, I need you to create an empty 'timelineStyle' function in `Timeline`, and pass it to `TimelineType` as the `style` function.  This will be like the `style` prop in `OrderDrawing.tsx` file.  We'll need the typings bits for this.
* [x] I wish to provide tooltips for created features. For this, I need to provide an implementation of `onEachFeature`. So, I'll need it added to `TimelineType`


Note: the above props are as defined for the [parent](https://leafletjs.com/reference.html#geojson) `L.GeoJSON` class:

**Comment by lebaphi** [comment]
> Handful of changes, please:
> 
> * [ ]  Don't display blue-line to left of thumb-marker. There is guidance on how to do it [here](https://www.w3schools.com/howto/howto_js_rangeslider.asp)
> * [ ]  I wish to color the features according to their Force. So, I need you to create an empty 'timelineStyle' function in `Timeline`, and pass it to `TimelineType` as the `style` function.  This will be like the `style` prop in `OrderDrawing.tsx` file.  We'll need the typings bits for this.
> * [ ]  I wish to provide tooltips for created features. For this, I need to provide an implementation of `onEachFeature`. So, I'll need it added to `TimelineType`
> 
> Note: the above props are as defined for the [parent](https://leafletjs.com/reference.html#geojson) `L.GeoJSON` class:

I'm working on it

**Comment by lebaphi** [comment]
Hi @IanMayo , I think we have a small issue with the first item
> Don't display blue-line to left of thumb-marker

it seems we need to use this css property `-webkit-appearance: none;` to override the background color of the slider, but it will lost the data list (small vertical lines) below the slider too
before:
image
after:
image

**Comment by IanMayo** [comment]
>  but it will lost the data list (small vertical lines) below the slider too

Oh dear, they look quite valuable.   They mark the times at which events occur.  Let's not hide the blue line then.

**Comment by lebaphi** [comment]
> > but it will lost the data list (small vertical lines) below the slider too
> 
> Oh dear, they look quite valuable. They mark the times at which events occur. Let's not hide the blue line then.

let me check if we can re-build the data-list in this case

**Comment by IanMayo** [comment]
> let me check if we can re-build the data-list in this case

No, it's ok.  Let's just leave the blue in-fill in place, and not use the `-webkit-appearance: none;` trick.

I am still interested in [these](features) though.

**Comment by lebaphi** [comment]
Hi @IanMayo , so for this one
> I wish to color the features according to their Force. So, I need you to create an empty 'timelineStyle' function in Timeline, and pass it to TimelineType as the style function. This will be like the style prop in OrderDrawing.tsx file. We'll need the typings bits for this.

What part do you want to style?, since we are using external library and it is compacted so we can't modify the javascript file, to style it we can only process in the `Timeline` component by using javascript.

**Comment by IanMayo** [comment]
Sure, don't worry.  We're _Going with the flow_ of what is meant to happen.

The TimelineData object extends Leaflet's GeoJSON class.  That class takes a `style` prop for a function that returns the styling to use for a feature: 
https://docs.eegeo.com/eegeo.js/v0.1.780/docs/leaflet/L.GeoJson/

I can't see where to specify it - so I'd like you to show me how to provide the function. You can just use an empty function called `timelineSyle`.  We do a similar thing in `OrderDrawing.tsx`.

File: packages/components/src/local/organisms/planning-channel/helpers/Timeline.tsx [hunk 1]
```diff
@@ -25,7 +25,12 @@ const Timeline: React.FC<TimelineProps> = ({ showControl, data }) => {
       setTimelineControl(timelineControl)
     }
     if (!timelineData && data) {
-      const timeline = L.timeline(data)
+      const timeline = L.timeline(data, {
```
Comment (IanMayo) [inline packages/components/src/local/organisms/planning-channel/helpers/Timeline.tsx:37]: perfect - just what I needed, thanks @lebaphi

File: packages/components/src/local/organisms/planning-channel/typings.d.ts [hunk 2]
```diff
@@ -66,7 +67,9 @@ type TimelineControlOption = {
 }
 
 type TimelineDataOption = {
-  pointToLayer: (data: any, latlng: L.LatLngExpression) => void
```
Comment (IanMayo) [inline packages/components/src/local/organisms/planning-channel/typings.d.ts]: I tried this several times, but couldn't get it right, thanks! 😍

**Pull request merged**